python/ray/serve/model_wrappers.py

```diff
@@ -101,7 +107,17 @@ async def batched_predict(inp):
 
     async def predict(self, inp):
         """Perform inference directly without HTTP."""
-        return await self.batched_predict(inp)
+        results = await self.batched_predict(inp)
```

Comment:
@simon-mo I'd like to avoid adjusting the `Predictor.predict()` methods, so we should be fully compatible with starlette from the serving wrapper side. Does this make sense? What is the return type of batched_predict? I expected a batch result, but it seems I only ever get scalars here. Please take a look at the conversion in lines 113ff

Reply:
hmm we shouldn't need to to something like this. Is the issue unpacking or serializability?

Reply:
i.e. what happen if we don't do the conversion? Serve have some default serializers actually.

Reply:
if you are getting this https://github.com/ray-project/ray/issues/24240, I can help dive deeper.